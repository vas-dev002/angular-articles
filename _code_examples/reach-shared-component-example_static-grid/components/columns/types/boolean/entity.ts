export interface BooleanGroupParams {
	value: string;
}
