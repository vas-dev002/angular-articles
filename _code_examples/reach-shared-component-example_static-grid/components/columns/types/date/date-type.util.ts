import * as moment from "moment";
import { isString, isEmptyStringifyDate } from "../../../../../../../queries/util/input";

export abstract class DateDasType {
	public static IsEmpty<T>(value: T): boolean {
		return !!value && value.toString() !== "null";
	}

	// generic T should be replaced by our custom Date entity type
	public static getValue<T>(value: T): string {
		return this.valueToFormattedString(value);
	}

	private static valueToFormattedString<T>(value: T): string {
		return isString(value) && !isEmptyStringifyDate(value) ? moment(value).format("ll") : "";
	}
}
