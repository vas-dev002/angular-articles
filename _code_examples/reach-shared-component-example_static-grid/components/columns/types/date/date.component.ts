import { Component } from "@angular/core";
import { ICellRendererAngularComp } from "@ag-grid-community/angular";
import { ICellRendererParams } from "@ag-grid-community/core";
import { DateDasType } from "./date-type.util";

@Component({
	selector: "date-sg",
	template: `<span>{{ getValue() }}</span>`
})
export class DateSGRendererComponent implements ICellRendererAngularComp {
	public params: ICellRendererParams;

	agInit(params: ICellRendererParams): void {
		this.params = params;
	}

	getValue(): string {
		const value = this.params.value?.date || this.params.value;

		return DateDasType.getValue(value);
	}

	refresh(): boolean {
		return false;
	}
}
