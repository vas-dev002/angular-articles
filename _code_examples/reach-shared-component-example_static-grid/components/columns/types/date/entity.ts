import { ValueFormatterParams } from "@ag-grid-community/core";

export interface IDateValue {
	date: string;
}

export interface IDateGroupKeyParams {
	value: string;
	date: string;
}

export interface IDateValueFormatterParams extends ValueFormatterParams {
	value: string | IDateValue;
}
