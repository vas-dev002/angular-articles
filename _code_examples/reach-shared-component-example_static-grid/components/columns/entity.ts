import { Type, Injector } from "@angular/core";
import { ICellRendererAngularComp } from "@ag-grid-community/angular";
import { ColDef } from "@ag-grid-community/core";
import { IColumnMetaSG } from "../main/entity";
import { StaticGridComponent } from "../main/comp";
import { ColumnDisplayType } from "./types";
import { IColDefAbstractFactory } from "./types/base/base-abstract-factory/models";
import { ObjectDetailsType } from "@evergreen-shared/entity";

export interface ITypeControllerSG {
	renderComponent: Type<ICellRendererAngularComp>;
	columnDefinition(): ColDef;
}

export interface IFilterOptionsSG {
	text: string;
	value: unknown;
	props?: Object;
	sortOrder?: number;
}

export interface ITypeControllerParams {
	meta: IColumnMetaSG;
	grid: StaticGridComponent;
	customOptions: Object;
	injector?: Injector;
}

export interface ILinkMeta {
	uRLColumnName: string;
	objectType: string;
	objectTypeColumnName: string;
	labelColumnName: string;
	keyColumnName: string;
	parameters: string;
	urlParameters: string[];
	renderLinkProperty: string;
}

export interface IOutcomeComparatorValue {
	status: number;
	text: string;
	tooltip: string;
}

export enum ApplicationForwardPathTypeEnum {
	None = 0,
	Retire = 1,
	Keep = 2,
	ForwardPath = 3,
	Uncategorized = 4
}

export enum ApplicationAssociationTypeEnum {
	None = 0,
	Dynamic = 1,
	StickyInclude = 2,
	StickyExclude = 3,
	ForwardPathed = 4,
	Mixed = 5
}

export enum StringValueGetterMode {
	String,
	SelectedItem,
	Params
}

export interface ITypedController {
	key: ColumnDisplayType;
	factory: Type<IColDefAbstractFactory>;
}

export enum TaskFlags {
	IsBucketTask = "IsBucketTask",
	IsReadOnlyTask = "IsReadOnlyTask",
	IsSystemTask = "IsSystemTask"
}

export interface IDateTimeEditorChanges {
	taskId: number;
	projectId: number;
	id: number;
	objectType: ObjectDetailsType;
	date: string;
	postBackParameters: { [key: string]: unknown };
}

export interface ICapacityDateTimeEditorChanges extends IDateTimeEditorChanges {
	slotId: number;
	slotName: string;
}
