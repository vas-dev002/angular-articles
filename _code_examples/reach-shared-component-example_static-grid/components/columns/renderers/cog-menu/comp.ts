import { Component } from "@angular/core";
import { ICellRendererAngularComp } from "@ag-grid-community/angular";
import { RowNode } from "@ag-grid-community/core";
import { ICellCogMenuParams, ICogMenuItem } from "./entity";

@Component({
	selector: "child-cell",
	templateUrl: "tpl.html"
})
export class CogMenuSGComponent implements ICellRendererAngularComp {
	public permissionStatus: boolean;
	public params: ICellCogMenuParams;
	public items: ICogMenuItem[];
	public isOpened = false;
	private _getIndex(): number {
		let index = this.params.rowIndex;
		this.params.api.forEachNodeAfterFilterAndSort((rowNode: RowNode, rowIndex: number) => {
			if (this.params.node.id === rowNode.id) {
				index = rowIndex;
			}
		});
		return index;
	}

	public agInit(params: ICellCogMenuParams): void {
		this.params = params;
		this.items = params.menuItems;
	}

	public refresh(): boolean {
		return false;
	}

	public show(item: ICogMenuItem) {
		return item.hide && item.hide({ params: this.params, rowIndex: this._getIndex() });
	}

	public onClick(item: ICogMenuItem) {
		item.onClick({ params: this.params, rowIndex: this._getIndex() });
	}

	public get isDisabled(): boolean {
		return !!this.params && !!this.params.isDisabled && this.params.isDisabled(this.params.data);
	}
}
