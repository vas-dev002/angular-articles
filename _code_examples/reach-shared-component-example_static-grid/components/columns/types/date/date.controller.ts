import { Type } from "@angular/core";
import { ICellRendererAngularComp } from "@ag-grid-community/angular";
import { ColDef, ITooltipParams, ValueFormatterParams } from "@ag-grid-community/core";

import { ITypeControllerSG } from "../../entity";
import { BaseTypeController } from "../base/base.controller";
import { DateSGRendererComponent } from "./date.component";
import { dateComparator } from "../../../../../util/column";
import { DateDasType } from "./date-type.util";
import { IDateValue, IDateGroupKeyParams } from "./entity";

/**
 * @deprecated. Use builder and factory instead.
 */
export class DateControllerSG extends BaseTypeController implements ITypeControllerSG {

	public renderComponent: Type<ICellRendererAngularComp> = DateSGRendererComponent;

	public columnDefinition(): ColDef {
		return {
			...this.baseDef
		};
	}

	protected customFilterDef(): ColDef {
		return {
			filter: "agDateColumnFilter",
			filterParams: {
				defaultOption: "equals",
				filterOptions: [
					"equals",
					"notEqual",
					"lessThanOrEqual",
					"greaterThan",
					"greaterThanOrEqual"
				],
				comparator: dateComparator,
				inRangeInclusive: false,
				suppressAndOrCondition: true
			}
		};
	}

	protected get customSortDef(): ColDef {
		return {
			sortingOrder: ["desc", "asc", null]
		};
	}

	protected getTooltip(): (params: ITooltipParams) => string {
		return (params: ITooltipParams) => this.transformDateValue(params);
	}

	protected getStringValue(): (params: ValueFormatterParams) => string {
		return (params: ValueFormatterParams) => this.transformDateValue(params);
	}

	protected getGroupKey(): <T>(params: { value: T } | IDateValue) => string {
		return (params: IDateGroupKeyParams) => {
			const typedValue = params?.date || params.value;

			return JSON.stringify({
				stringValueMode: true,
				value: DateDasType.getValue(typedValue)
			});
		};
	}

	private readonly transformDateValue = (params: ValueFormatterParams | ITooltipParams) => {
		const value = params.value?.date || params.value;
		return DateDasType.getValue(value);
	}
}
