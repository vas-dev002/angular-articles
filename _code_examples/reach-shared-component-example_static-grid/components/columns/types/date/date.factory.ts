import { ColDef } from "@ag-grid-community/core";
import { ITypeControllerParams } from "../../entity";

import { IColDefAbstractFactory } from "../base/base-abstract-factory/models";
import { DateControllerSG } from "./date.controller";

export class DateFactorySG implements IColDefAbstractFactory {

	createColDef(controllerParams: ITypeControllerParams): ColDef {
		const controller = new DateControllerSG(controllerParams);

		return controller.columnDefinition();
	}
}
