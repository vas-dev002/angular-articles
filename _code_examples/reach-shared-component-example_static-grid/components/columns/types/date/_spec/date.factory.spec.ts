import { ColDef } from "@ag-grid-community/core";
import { Injector } from "@angular/core";
import { TranslateModule, TranslateService } from "@ngx-translate/core";

import { inject, TestBed } from "@angular/core/testing";

import { IColumnMetaSG } from "../../../../main/entity";
import { StaticGridComponent } from "../../../../main/comp";
import { COMMON_META } from "../../base/_spec/common.set";
import { DateFactorySG } from "../date.factory";

declare const expect: jest.Expect;

describe("Static Grid Column Type Compliance: ", () => {
	const meta: IColumnMetaSG = COMMON_META;
	const customOptions = {};
	const rowData = [];
	const grid = { rowData } as StaticGridComponent;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [TranslateModule.forRoot()],
			providers: [TranslateService]
		});
	});

	describe(`Controller: DateFactorySG `, () => {
		let factory: DateFactorySG;
		let columnDefinition: ColDef;
		let injector: Injector;

		beforeEach(inject([Injector], (inj: Injector) => {
			injector = inj;
			factory = new DateFactorySG();
		}));

		describe("[Method]: createColDef ", () => {
			it("should return default column definition", () => {
				columnDefinition = factory.createColDef({ meta, grid, customOptions, injector });
				expect(columnDefinition).toMatchSnapshot();
			});
		});
	});
});
