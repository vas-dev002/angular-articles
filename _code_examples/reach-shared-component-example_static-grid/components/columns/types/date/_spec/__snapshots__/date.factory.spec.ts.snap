// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Static Grid Column Type Compliance:  Controller: DateFactorySG  [Method]: createColDef  should return default column definition 1`] = `
Object {
  "cellClass": "testCss",
  "cellClassRules": Object {
    "sg-cell-disabled": [Function],
  },
  "cellRendererFramework": [Function],
  "field": "testColumn",
  "filter": "agDateColumnFilter",
  "filterParams": Object {
    "comparator": [Function],
    "defaultOption": "equals",
    "filterOptions": Array [
      "equals",
      "notEqual",
      "lessThanOrEqual",
      "greaterThan",
      "greaterThanOrEqual",
    ],
    "inRangeInclusive": false,
    "suppressAndOrCondition": true,
  },
  "headerName": "Test Header",
  "hide": true,
  "keyCreator": [Function],
  "lockVisible": false,
  "maxWidth": 500,
  "menuTabs": Array [
    "generalMenuTab",
    "filterMenuTab",
    "columnsMenuTab",
  ],
  "minWidth": 50,
  "sortingOrder": Array [
    "desc",
    "asc",
    null,
  ],
  "tooltipComponentFramework": [Function],
  "tooltipField": undefined,
  "tooltipValueGetter": [Function],
  "valueFormatter": [Function],
}
`;
